src/etc/inc/interfaces.inc

```diff
@@ -6799,6 +6804,72 @@ function get_wireless_modes($interface) {
 	return($wireless_modes);
 }
 
+function get_wireless_ht_modes($interface) {
+	/* return wireless HT */
+	$wireless_hts_supported = array(0 => gettext('Auto'));
+
+	$cloned_interface = get_real_interface($interface);
+
+	if ($cloned_interface && is_interface_wireless($cloned_interface)) {
+		$chan_list = "/sbin/ifconfig -v {$cloned_interface} list chan";
+		$stack_list = "/usr/bin/awk -F\"Channel \" '{ gsub(/\\*/, \" \"); print \$2 \"\\\n\" \$3 }'";
+		$format_list = "/usr/bin/awk '{print \$5 \",\" \$6 \",\" \$1}'";
+
+		$interface_channels = "";
+		exec("$chan_list | $stack_list | sort -u | $format_list 2>&1", $interface_channels);
+
+		foreach ($interface_channels as $channel) {
+			$channel_line = explode(",", $channel);
+			$wireless_ht = trim($channel_line[1]);
+			if (!empty($wireless_ht)) {
+				$wireless_hts_supported[$wireless_ht] = strtoupper($wireless_ht);
+			}
+		}
+	}
+	return($wireless_hts_supported);
+}
+
+function get_wireless_ht_list($interface) {
+	/* return wireless HT */
+	$wireless_hts = array();
+
+	$cloned_interface = get_real_interface($interface);
+
+	if ($cloned_interface && is_interface_wireless($cloned_interface)) {
+		$chan_list = "/sbin/ifconfig -v {$cloned_interface} list chan";
+		$stack_list = "/usr/bin/awk -F\"Channel \" '{ gsub(/\\*/, \" \"); print \$2 \"\\\n\" \$3 }'";
+		$format_list = "/usr/bin/awk '{print \$5 \",\" \$6 \",\" \$1}'";
+
+		$interface_channels = "";
+		exec("$chan_list | $stack_list | sort -u | $format_list 2>&1", $interface_channels);
```

Comment:
Why repeat all of this? Make a common function and call it from both places.

Reply:
all fixed